keyboards/macroflow_original/keymaps/default/keymap.c

```diff
@@ -0,0 +1,297 @@
+        KC_1, KC_2, KC_3, KC_4,
+        KC_5, KC_6, KC_7, KC_8
+    ),
+    [1] = LAYOUT_macroflow_2x4(
+        KC_9, KC_CYCLE_LAYERS,
+        KC_A, KC_B, KC_C, KC_D,
+        KC_E, KC_F, KC_G, KC_H
+    ),
+    [2] = LAYOUT_macroflow_2x4(
+        KC_I, KC_CYCLE_LAYERS,
+        KC_J, KC_K, KC_L, KC_M,
+        KC_N, KC_O, KC_P, KC_Q
+    ),
+    [3] = LAYOUT_macroflow_2x4(
+        KC_R, KC_CYCLE_LAYERS,
+        KC_S, KC_T, KC_U, KC_V,
+        KC_W, KC_X, KC_Y, KC_Z
+    ),
+    [4] = LAYOUT_macroflow_2x4(
+        KC_TRNS, KC_CYCLE_LAYERS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+    [5] = LAYOUT_macroflow_2x4(
+        KC_TRNS, KC_CYCLE_LAYERS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+    [6] = LAYOUT_macroflow_2x4(
+        KC_TRNS, KC_CYCLE_LAYERS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+    [7] = LAYOUT_macroflow_2x4(
+        KC_TRNS, KC_CYCLE_LAYERS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    )
+};
+
+// 1st layer on the cycle
+#define LAYER_CYCLE_START 0
+// Last layer on the cycle
+#define LAYER_CYCLE_END 8
+
+// Add the behaviour of this new keycode
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+    case KC_CYCLE_LAYERS:
+      // Our logic will happen on presses, nothing is done on releases
+      if (!record->event.pressed) { 
+        // We've already handled the keycode (doing nothing), let QMK know so no further code is run unnecessarily
+        return false;
+      }
+
+      uint8_t current_layer = get_highest_layer(layer_state);
+
+      // Check if we are within the range, if not quit
+      if (current_layer > LAYER_CYCLE_END || current_layer < LAYER_CYCLE_START) {
+        return false;
+      }
+
+      uint8_t next_layer = current_layer + 1;
+      if (next_layer > LAYER_CYCLE_END) {
+          next_layer = LAYER_CYCLE_START;
+      }
+      layer_move(next_layer);
+      return false;
+
+    // Process other keycodes normally
+    default:
+      return true;
+  }
+}
+
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
+    [0] = { ENCODER_CCW_CW(KC_MS_WH_UP, KC_MS_WH_DOWN)  },
+    [1] = { ENCODER_CCW_CW(KC_MS_WH_UP, KC_MS_WH_DOWN)  },
+    [2] = { ENCODER_CCW_CW(KC_MS_WH_UP, KC_MS_WH_DOWN)  },
+    [3] = { ENCODER_CCW_CW(KC_MS_WH_UP, KC_MS_WH_DOWN)  },
+    [4] = { ENCODER_CCW_CW(KC_MS_WH_UP, KC_MS_WH_DOWN)  },
+    [5] = { ENCODER_CCW_CW(KC_MS_WH_UP, KC_MS_WH_DOWN)  },
+    [6] = { ENCODER_CCW_CW(KC_MS_WH_UP, KC_MS_WH_DOWN)  },
+    [7] = { ENCODER_CCW_CW(KC_MS_WH_UP, KC_MS_WH_DOWN)  }
+};
+#endif
+
+#ifdef OLED_ENABLE
+
+#ifndef SHOW_LOGO
+#    define SHOW_LOGO 2500
+#endif
+static uint16_t oled_logo_timer = 0;
+static bool has_startup_ran = false;
+
+bool oled_task_user(void) {
```

Comment:
All of this oled code should be moved to `macroflow_original.c` and switched to the *_kb versions of the functions.

Reply:
Thank you, drashna, but what do you mean by switching to the *_kb versions of the functions?

Reply:
```c
bool oled_task_kb(void) {
    if (!oled_task_user()) {
        return false;
    }
    // code
    return true;
}
```